src/json-patch/codec/binary/encodeOperationToMsgpack.ts

```diff
@@ -0,0 +1,308 @@
+import type {IMessagePackEncoder} from '@jsonjoy.com/json-pack/lib/msgpack';
+import type {AbstractOp} from '../../op/AbstractOp';
+import {OPCODE} from '../../constants';
+
+/**
+ * Standalone function to encode operations to MessagePack format.
+ * This replaces the instance method approach with a centralized switch statement.
+ */
+export function encodeOperationToMsgpack(op: AbstractOp, encoder: IMessagePackEncoder, parent?: AbstractOp): void {
```

Comment:
- Rename this function to `encodeOp()`.
- And, move it inside the `Encoder` class in `src/json-patch/codec/binary/Encoder.ts` file.